pkg/helpers/ssh.go

```diff
@@ -129,6 +129,35 @@ func (sshPayload *SSHPayload) SSHExecute(storage resources.StorageFactory, log r
 				return log.NewError("unsupported key type: %s", keyType)
 			})}
 
+	if provider == consts.CloudAws {
+		config = &ssh.ClientConfig{
+			User: "ubuntu",
+			Auth: []ssh.AuthMethod{
+				ssh.PublicKeys(signer),
+			},
+
+			HostKeyAlgorithms: []string{
+				ssh.KeyAlgoECDSA256,
+				ssh.KeyAlgoED25519,
+				// ssh.KeyAlgoRSA,
+			},
+			HostKeyCallback: ssh.HostKeyCallback(
+				func(hostname string, remote net.Addr, key ssh.PublicKey) error {
+					actualFingerprint := ssh.FingerprintSHA256(key)
+					keyType := key.Type()
+					if keyType == ssh.KeyAlgoRSA || keyType == ssh.KeyAlgoDSA || keyType == ssh.KeyAlgoECDSA256 || keyType == ssh.KeyAlgoED25519 {
+						expectedFingerprint, err := returnServerPublicKeys(sshPayload.PublicIP, provider)
+						if err != nil {
+							return err
+						}
+						if expectedFingerprint != actualFingerprint {
+							return fmt.Errorf("[ssh] mismatch of fingerprint")
+						}
+						return nil
+					}
+					return fmt.Errorf("[ssh] unsupported key type: %s", keyType)
+				})}
+	}
```

Comment:
I think we can merge it with the previous implementation.just as we discussed
we can use the keytype as a function argument for the `returnServerPublicKeys`